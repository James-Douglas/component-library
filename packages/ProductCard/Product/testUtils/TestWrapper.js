import React from 'react';
import ProductProvider from '../../Context/ProductProvider';

const ProductTestWrapper = ({
  // eslint-disable-next-line react/prop-types
  children, product, onDetails = () => {}, onSelect = () => {},
}) => (
  <ProductProvider product={product} onDetails={onDetails} onSelect={onSelect}>
    { children }
  </ProductProvider>
);

export default ProductTestWrapper;
