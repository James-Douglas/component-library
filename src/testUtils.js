// test-utils.js
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './providers/ManorProvider';

const customRender = (ui, options) => render(ui, { wrapper: ManorProvider, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
