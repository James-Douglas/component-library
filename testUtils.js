import React from 'react';
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './packages/Provider/ManorProvider';

const ManorTestWrapper = ({ children }) => (
  <ManorProvider>
    {children}
  </ManorProvider>
);

const customRender = (ui, options) => render(ui, { wrapper: ManorTestWrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
