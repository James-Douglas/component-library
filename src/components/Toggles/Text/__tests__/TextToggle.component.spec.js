import React from 'react';
import { render } from '../../../../testUtils';
import TextToggle from '../TextToggle.component';

describe('TextToggle', () => {
  it('renders with props', () => {
    const { container } = render(<TextToggle id="test" value="test" label="hello" />);
    expect(container.innerHTML).toMatchSnapshot();
  });
});
