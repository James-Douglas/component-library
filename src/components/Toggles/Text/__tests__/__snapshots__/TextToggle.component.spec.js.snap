// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextToggle renders with props 1`] = `"<div class=\\"sc-AxjAm kBRMiC\\"><input tabindex=\\"0\\" id=\\"test\\" type=\\"radio\\" name=\\"\\" class=\\"sc-AxirZ exDnLo\\" value=\\"test\\"><label for=\\"test\\" class=\\"sc-AxiKw ikDIEa\\"><div class=\\"sc-AxhCb bxBvvw\\"><div class=\\"sc-AxhUy gsFBrZ\\"><div class=\\"sc-AxgMl inBFXj\\"><div class=\\"sc-AxheI hRqFao\\">hello</div></div></div></div></label></div>"`;
